### ISSUE TITLE
Crash in GraphQLExecutor.swift line 143

### ISSUE THREAD

**Issue by Dimillian** [apollographql/apollo-ios#2217]
## Bug report

We’ve updated our App to the 0.51.1 version, and we see a new crash since we shipped this update. 

```
GraphQLExecutor.swift line 143
GraphQLExecutor.execute<A>(selections:on:info:accumulator:)
```

## Versions

Please fill in the versions you're currently using: 

- `apollo-ios`: 0.51.1
- Xcode version: 13.2.1
- Swift version: 5.5
- Package manager:SPM

## Steps to reproduce

I have no idea, Crahslytics say it happens very early in the lifecyle of the app, and from our logs, it’s not clear if a specific query trigger this. 

## Further details

Here is the log

```
Crashed: NSOperationQueue 0x10264e1f0 (QOS: USER_INITIATED)
0  libsystem_kernel.dylib         0x6bbc __pthread_kill + 8
1  libsystem_pthread.dylib        0xd854 pthread_kill + 208
2  libsystem_c.dylib              0x1f6a4 abort + 124
3  libswiftCore.dylib             0x2ed01c swift::warningv(unsigned int, char const*, char*) + 138
4  libswiftCore.dylib             0x2f6698 swift::MetadataAllocator::Allocate(unsigned long, unsigned long) + 516
5  libswiftCore.dylib             0x2f9970 std::__1::pair<(anonymous namespace)::TupleCacheEntry*, swift::MetadataResponse> swift::LockingConcurrentMap<(anonymous namespace)::TupleCacheEntry, (anonymous namespace)::TupleCacheStorage>::getOrInsert<(anonymous namespace)::TupleCacheEntry::Key, swift::MetadataRequest&, swift::TargetValueWitnessTable<swift::InProcess> const*&>((anonymous namespace)::TupleCacheEntry::Key, swift::MetadataRequest&, swift::TargetValueWitnessTable<swift::InProcess> const*&) + 2920
6  libswiftCore.dylib             0x2f8bd0 swift_getTupleTypeMetadata + 104
7  libswiftCore.dylib             0x315cb0 swift::Demangle::__runtime::TypeDecoder<(anonymous namespace)::DecodedMetadataBuilder>::decodeMangledType(swift::Demangle::__runtime::Node*, unsigned int, bool) + 12812
8  libswiftCore.dylib             0x312024 swift_getTypeByMangledNodeImpl(swift::MetadataRequest, swift::Demangle::__runtime::Demangler&, swift::Demangle::__runtime::Node*, void const* const*, std::__1::function<swift::TargetMetadata<swift::InProcess> const* (unsigned int, unsigned int)>, std::__1::function<swift::TargetWitnessTable<swift::InProcess> const* (swift::TargetMetadata<swift::InProcess> const*, unsigned int)>) + 552
9  libswiftCore.dylib             0x311d9c swift_getTypeByMangledNode + 540
10 libswiftCore.dylib             0x312510 swift_getTypeByMangledNameImpl(swift::MetadataRequest, __swift::__runtime::llvm::StringRef, void const* const*, std::__1::function<swift::TargetMetadata<swift::InProcess> const* (unsigned int, unsigned int)>, std::__1::function<swift::TargetWitnessTable<swift::InProcess> const* (swift::TargetMetadata<swift::InProcess> const*, unsigned int)>) + 968
11 libswiftCore.dylib             0x30fd14 swift_getTypeByMangledName + 540
12 libswiftCore.dylib             0x301260 swift_getAssociatedTypeWitnessSlowImpl(swift::MetadataRequest, swift::TargetWitnessTable<swift::InProcess>*, swift::TargetMetadata<swift::InProcess> const*, swift::TargetProtocolRequirement<swift::InProcess> const*, swift::TargetProtocolRequirement<swift::InProcess> const*) + 388
13 libswiftCore.dylib             0x2fffb4 swift_getAssociatedTypeWitness + 180
14 hangtag                        0x4f2e8c GraphQLExecutor.execute<A>(selections:on:info:accumulator:) + 143 (GraphQLExecutor.swift:143)
15 hangtag                        0x4f29b8 GraphQLExecutor.execute<A>(selections:on:withKey:variables:accumulator:) + 140 (GraphQLExecutor.swift:140)
16 hangtag                        0x4fd0a0 GraphQLResponse.parseResult(cacheKeyForObject:) + 49 (GraphQLResponse.swift:49)
17 hangtag                        0x50d0dc JSONResponseParsingInterceptor.parseResult<A>(from:cachePolicy:) + 83 (JSONResponseParsingInterceptor.swift:83)
18 hangtag                        0x50ce48 JSONResponseParsingInterceptor.interceptAsync<A>(chain:request:response:completion:) + 60 (JSONResponseParsingInterceptor.swift:60)
19 hangtag                        0x51d368 RequestChain.proceedAsync<A>(request:response:completion:) + 92 (RequestChain.swift:92)
20 hangtag                        0x520004 specialized ResponseCodeInterceptor.interceptAsync<A>(chain:request:response:completion:) + 55 (ResponseCodeInterceptor.swift:55)
21 hangtag                        0x51d368 RequestChain.proceedAsync<A>(request:response:completion:) + 92 (RequestChain.swift:92)
22 hangtag                        0x5117d8 closure #1 in NetworkFetchInterceptor.interceptAsync<A>(chain:request:response:completion:) + 58 (NetworkFetchInterceptor.swift:58)
23 hangtag                        0x5119d4 partial apply for closure #1 in NetworkFetchInterceptor.interceptAsync<A>(chain:request:response:completion:) + 4300806612 (<compiler-generated>:4300806612)
24 hangtag                        0x522bd8 specialized URLSessionClient.urlSession(_:task:didCompleteWithError:) + 218 (URLSessionClient.swift:218)
25 hangtag                        0x5217e8 @objc URLSessionClient.urlSession(_:task:didCompleteWithError:) + 4300871656 (<compiler-generated>:4300871656)
26 CFNetwork                      0x19df0 CFURLRequestCopyHTTPRequestMethod + 2212
27 Foundation                     0x4e26c __NSBLOCKOPERATION_IS_CALLING_OUT_TO_A_BLOCK__ + 16
28 Foundation                     0x5f920 -[NSBlockOperation main] + 100
29 Foundation                     0x39a8c __NSOPERATION_IS_INVOKING_MAIN__ + 20
30 Foundation                     0x49f20 -[NSOperation start] + 764
31 Foundation                     0x4d368 __NSOPERATIONQUEUE_IS_STARTING_AN_OPERATION__ + 20
32 Foundation                     0x5ab7c __NSOQSchedule_f + 180
33 libdispatch.dylib              0x46048 _dispatch_block_async_invoke2 + 104
34 libdispatch.dylib              0x64094 _dispatch_client_callout + 16
35 libdispatch.dylib              0x3bab8 _dispatch_continuation_pop$VARIANT$armv81 + 432
36 libdispatch.dylib              0x3b1f4 _dispatch_async_redirect_invoke + 584
37 libdispatch.dylib              0x48630 _dispatch_root_queue_drain + 328
38 libdispatch.dylib              0x48dec _dispatch_worker_thread2 + 160
39 libsystem_pthread.dylib        0x1dd4 _pthread_wqthread + 224
40 libsystem_pthread.dylib        0x193c start_wqthread + 8
```

**Comment by calvincestari** [comment]
Hi @Dimillian 👋🏻 - what version are you updating from? `0.51.2` was released shortly after `0.51.1` to fix a problem with APQs. Might be worth trying that too to rule out any other side-effect.

**Comment by Dimillian** [comment]
Hello @calvincestari from 0.49. Will update it to latest.

**Comment by sergiymomot** [comment]
The issue is with swift::MetadataAllocator
It seems broken in iOS 15.4 specifically
We saw the same top of the stack trace with other SDKs crashes

There is a bug reported here: https://bugs.swift.org/plugins/servlet/mobile#issue/SR-16047

At this point it’s unknown if it’s a Swift or iOS issue, but in any case probably only new iOS update will resolve it, so we have to wait

**Comment by calvincestari** [comment]
Thanks for sharing that @sergiymomot.

@Dimillian does that line up with your findings; crash on iOS 15.4 vs. other iOS versions?

**Comment by Dimillian** [comment]
It does match, it seems to be only on iOS 15.4.0.

**Comment by calvincestari** [comment]
The [PR](https://github.com/apple/llvm-project/pull/3917) referenced in that [Swift bug report](https://bugs.swift.org/plugins/servlet/mobile#issue/SR-16047) was merged back in Feb but unfortunately [Mike Ash just confirmed](https://bugs.swift.org/browse/SR-16047?focusedCommentId=65468&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-65468) that the 13.3 SDK won't help - the Swift runtime needs the fix. 😢

**Comment by calvincestari** [comment]
This appears to have been resolved with iOS 15.4.1 - if you're still having an issue please let us know.
